### PR TITLE
Show page for individual approaches

### PR DESCRIPTION
**Pull request by dem4ron**
## UI changes:

Screenshot 2022-10-13 at 14 19 57

Screenshot 2022-10-13 at 14 20 42

### CODE REVIEW

File: app/controllers/tracks/approaches_controller.rb [hunk 1]
```diff
@@ -23,9 +23,11 @@ def show
...
         avatar_url: "https://avatars.githubusercontent.com/u/135246?v=4",
-        handle: "Erik",
+        handle: "ErikSchierboom",
+        name: "Erik Schierboom",
         formatted_reputation: 999
       ),
+      content_html: "<h1>Hello</h1>",
       title: "Utilising In-built Date Functions for Consistency",
       published_at: DateTime.now
     )
```
Comment (iHiD) [inline app/controllers/tracks/approaches_controller.rb]: @ErikSchierboom I presume this shouldn't be hardcoded? :)

File: app/helpers/react_components/track/credits.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+      def to_s
+        super("track-credits", {
+          avatar_urls:,
+          top_count:,
+          top_label:,
+          bottom_count:,
+          bottom_label:,
+          max:
+        }
```
Comment (iHiD) [inline app/helpers/react_components/track/credits.rb]: You probably don't need the `{` and `}`, but if you do, then this one and the line below should be on the same line (to match how they are on L7)

File: app/views/tracks/approaches/show.html.haml [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
-      .py-8.px-24.bg-gray.border-2.border-gradient.rounded-100.flex.flex-row.items-center{ class: "h-[68px]" }
-        = graphical_icon "balloons", category: :graphics, css_class: "h-[40px] mr-12"
-        = graphical_icon "building", category: :graphics, css_class: "h-48 mr-12"
-        -# = track_icon css_class: "h-[40px] w-[40px] mr-12"
-          . = exercise_icon "amusement-park", css_class: "h-48 mr-12"
-        .flex.flex-col
-          .text-h5.whitespace-nowrap Amusement park
-          .textColor-6.font-normal.leading-150.text-16 Csharp
+      = link_to Exercism::Routes.track_exercise_path(@track, @exercise) do
+        .py-8.px-24.bg-gray.border-2.border-gradient-bgGray.rounded-100.flex.flex-row.items-center{ class: "h-[68px]" }
```
Comment (iHiD) [inline app/views/tracks/approaches/show.html.haml]: Merge these two lines. If you ever have a single div within a link, then normally the div's classes can be inlined into the link.

**Comment by iHiD** [comment]
This is looking really good, @dem4ron!

File: app/controllers/tracks/approaches_controller.rb (hunk 1 above)
Comment (dem4ron) [reply app/controllers/tracks/approaches_controller.rb]: this was me :) I just used it as dummy data to show something.

File: app/views/tracks/approaches/show.html.haml (hunk 3 above)
Comment (dem4ron) [reply app/views/tracks/approaches/show.html.haml]: true!

File: app/assemblers/assemble_exercise_approach_share_panel.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class AssembleExerciseApproachSharePanel
```
Comment (iHiD) [inline app/assemblers/assemble_exercise_approach_share_panel.rb]: This should be a serializer, not an assembler - there's no assembling happening.

This should be `SerializeApproach`.

Comment (iHiD) [inline app/assemblers/assemble_exercise_approach_share_panel.rb]: This should be a serializer, not an assembler - there's no assembling happening.

This should be `SerializeApproach`.

File: app/commands/git/sync_exercise_approach_introduction.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class Git::SyncExerciseApproachIntroduction
+  include Mandate
+
+  initialize_with :exercise, :config
+
+  def call
+    exercise.update!(
```
Comment (iHiD) [inline app/commands/git/sync_exercise_approach_introduction.rb:7]: Is this update actually needed? It feels like the `Create` below is actually doing the work of creating. What extra does this do?

File: app/commands/user/combine_authors_and_contributors.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class User::CombineAuthorsAndContributors
```
Comment (iHiD) [inline app/commands/user/combine_authors_and_contributors.rb]: The doesn't feel like a command to me as it doesn't actually do anything mutating, and it doesn't feel like it's part of the `User` scope. I'll have a think about where it should go.

File: app/commands/git/sync_exercise_approaches.rb [hunk 7]
```diff
@@ -8,47 +8,23 @@ def initialize(exercise)
     end
 
     def call
+      exercise.update(approaches:)
```
Comment (iHiD) [inline app/commands/git/sync_exercise_approaches.rb:12]:
```suggestion
      # This removes any approaches that aren't read from the config below
      exercise.update(approaches:)
```

File: app/commands/user/combine_authors_and_contributors.rb (hunk 6 above)
Comment (ErikSchierboom) [reply app/commands/user/combine_authors_and_contributors.rb]: Yeah I know. Maybe an assembler?

**Review by iHiD — APPROVED** [review]

**Pull request merged**